modules/assignment/src/main/java/com/privacy/wasabi/assignment/MongoAssignmentIngestionExecutor.java

```diff
@@ -0,0 +1,92 @@
+package com.privacy.wasabi.assignment;
+
+import com.intuit.wasabi.assignment.AssignmentIngestionExecutor;
+import com.intuit.wasabi.assignmentobjects.AssignmentEnvelopePayload;
+import com.intuit.wasabi.assignmentobjects.User;
+import com.intuit.wasabi.assignmentobjects.User.ID;
+import com.intuit.wasabi.experimentobjects.Bucket;
+import com.intuit.wasabi.experimentobjects.Experiment;
+import com.intuit.wasabi.experimentobjects.Experiment.Label;
+
+import com.mongodb.MongoClient;
+import com.mongodb.MongoClientURI;
+import com.mongodb.DB;
+import com.mongodb.DBCollection;
+import com.mongodb.BasicDBObject;
+import com.mongodb.ServerAddress;
+import com.mongodb.MongoException;
+
+import org.slf4j.Logger;
+import static org.slf4j.LoggerFactory.getLogger;
+
+import java.util.Map;
+import java.util.HashMap;
+import java.util.concurrent.Future;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.Date;
+
+import java.net.UnknownHostException;
+
+public class MongoAssignmentIngestionExecutor implements AssignmentIngestionExecutor {
+
+    public static final String NAME = "MONGOINGESTOR";
+    private static final Logger LOGGER = getLogger(MongoAssignmentIngestionExecutor.class);
+
+    private ExecutorService executor = Executors.newSingleThreadExecutor();
+    private MongoClient mongoClient = null;
+
+    private String mongoURI = System.getenv("MONGO_URI");
+    private String mongoDB = System.getenv("MONGO_DB");
+
+    @Override
+    public Future<?> execute(AssignmentEnvelopePayload assignmentEnvelopePayload) {
+        return executor.submit(() -> {
+            if (mongoURI != null && mongoDB != null) {
+                try {
+                    if (mongoClient == null || mongoClient.getDatabaseNames().isEmpty()) {
```

Comment:
So this change successfully shares the connection from assignment to assignment, but the first time assignments are pulled there can be a race condition where multiple connections get opened (in my testing, it pulls 5 assignments at the start and ended up opening 2 mongo connections in that process instead of just 1). Does that seem fine, or should I try to make sure only one connection ever gets opened?